libyul/optimiser/Suite.cpp

```diff
@@ -267,44 +267,29 @@ map<char, string> const& OptimiserSuite::stepAbbreviationToNameMap()
 	return lookupTable;
 }
 
+
+
 void OptimiserSuite::validateSequence(string const& _stepAbbreviations)
 {
+	stack<char> s;
```

Comment:
For validation you do not really need a stack. A simple counter tracking the current nesting level would be enough. Start with 0 and increase it when you encounter `[`, decrease it when you encounter `]`. If at any moment the counter gets negative, nesting is wrong (too many opening brackets). And if you reach the end of the string with a non-zero counter, there are too many closing brackets.

You will only need a stack in `runSequence()`.